### DISCUSSION TITLE
Do you have any plan to create monorepo?

### DISCUSSION THREAD

**Discussion by Rafat97** [foyzulkarim/mernboilerplate-antd#68] [Ideas]
@foyzulkarim . Thank you for this awesome open source project. 

I saw your project have three major components.

1. [client-pro](https://github.com/foyzulkarim/mern-boilerplate/tree/develop/client-pro)
2. [server](https://github.com/foyzulkarim/mern-boilerplate/tree/develop/server)
3. mongoDB

I think, If your project have this type of folder architecture. It will more convenient & structured. Also, it's easy to understandably the main components of the project.

```bash
root_dir
  |-- src
  |    |-- client-pro
  |    |      |-- ...
  |    |      |-- docker-compose.yml
  |    |      |-- package.json
  |    |-- server
  |    |      |-- ...
  |    |      |-- docker-compose.yml
  |    |      |-- package.json
  |    |-- mongoDB
  |    |      |-- ...
  |    |      |-- docker-compose.yml
  |    |      |-- package.json
  |-- package.json
  |-- lerna.json
```

root directory `package.json` file just run. lerna commends. Ex;
```
"start:docker": "lerna run start:docker-compose --stream && docker ps"
"stop:docker": "lerna run stop:docker-compose --stream && docker ps"
```

After doing `lerna` configuration. Just run `yarn start:docker` commend. And, It will spin up the whole project.

If you have plan on this monorepo. I can help on it. 

Thank you.

**Comment by foyzulkarim** [comment]
Brother. Thanks for the suggestion. Because this is boilerplate, I don't want to make things too opinionated. If possible, can you make such a way that both of the ways can work according to the preference of the dev? Maybe also we need to update the `readme` file to instruct the fellow dev about these two different ways.
